core/src/main/java/org/elasticsearch/action/admin/cluster/health/ClusterHealthResponse.java

```diff
@@ -214,12 +137,25 @@ public boolean isTimedOut() {
         return this.timedOut;
     }
 
+    public void setTimedOut(boolean timedOut) {
+        this.timedOut = timedOut;
+    }
+
     public ClusterHealthStatus getStatus() {
-        return status;
+        return clusterHealthStatusOverride != null ? clusterHealthStatusOverride : clusterHealth.getStatus();
```

Comment:
I think it will simpler to have a status variable and set it in the constructor to whatever clusterHealth said. Then this becomes a simple getter and setClusterHealthStatusOverride becomes a simple setter (setStatus)